Lib/http/cookiejar.py

```diff
@@ -641,7 +663,20 @@ def eff_request_host(request):
 
     """
     erhn = req_host = request_host(request)
-    if "." not in req_host:
+    if req_host.startswith('[') and req_host.endswith(']'):
+        from ipaddress import IPv6Address
+        try:
+            IPv6Address(req_host[1:-1])
+        except ValueError:
+            is_ipV6 = False
+        else:
+            is_ipV6 = True
+    else:
+        is_ipV6 = False
+    if "." not in req_host and not is_ipV6:
+        # avoid adding .local at the end of a IPV6 address
+        # for the additional ".local", see RFC 2965 section 1
+        # https://www.rfc-editor.org/rfc/rfc2965
```

Comment:
```suggestion
        # Avoid adding .local at the end of a IPV6 address.
        # See RFC 2965 [1] for the rationale of ".local".
        # [1]: https://www.rfc-editor.org/rfc/rfc2965
```

That way we can expand the comment in the future as much as we want.